keras_cv/layers/preprocessing/solarization_test.py

```diff
@@ -12,72 +12,71 @@
-            {"input_value": 127, "expected_output_value": 128},
-            {"input_value": 191, "expected_output_value": 64},
-            {"input_value": 255, "expected_output_value": 0},
-        ]
+        self._test_input_output(
+            layer=solarization,
+            input_value=input_value,
+            expected_value=expected_value,
+            dtype=tf.uint8,
+        )
 
-        for parameters in test_parameters:
-            self._test_input_output(
-                layer=solarization,
-                input_value=parameters["input_value"],
-                expected_value=parameters["expected_output_value"],
-                dtype=tf.uint8,
-            )
+    @parameterized.named_parameters(
+        ("0_245", 0, 245),
+        ("255_0", 255, 0),
+    )
+    def test_solarization_with_addition(self, input_value, output_value):
+        solarization = Solarization(addition=10.0)
 
-    @staticmethod
-    def _test_input_output(layer, input_value, expected_value, dtype):
-        dummy_input = tf.ones(shape=(2, 224, 224, 3), dtype=dtype) * input_value
-        expected_output = tf.ones(shape=(2, 224, 224, 3), dtype=dtype) * expected_value
-
-        output = layer(dummy_input)
-
-        tf.debugging.assert_equal(output, expected_output)
+        self._test_input_output(
+            layer=solarization,
+            input_value=input_value,
+            expected_value=output_value,
+            dtype=tf.float32,
+        )
 
-    def test_only_values_above_threshold_are_solarized_if_threshold_specified(self):
+    @parameterized.named_parameters(
+        ("0_0", 0, 0),
+        ("64_64", 64, 64),
+        ("127_127", 127, 127),
+        ("191_64", 191, 64),
+        ("255_0", 255, 0),
+    )
+    def test_only_values_above_threshold_are_solarized(self, input_value, output_value):
         solarization = Solarization(threshold=128)
 
-        test_parameters = [
-            {"input_value": 0, "expected_output_value": 0},
-            {"input_value": 64, "expected_output_value": 64},
-            {"input_value": 127, "expected_output_value": 127},
-            {"input_value": 191, "expected_output_value": 64},
-            {"input_value": 255, "expected_output_value": 0},
-        ]
+        self._test_input_output(
+            layer=solarization,
+            input_value=input_value,
+            expected_value=output_value,
+            dtype=tf.uint8,
+        )
 
-        for parameters in test_parameters:
-            self._test_input_output(
-                layer=solarization,
-                input_value=parameters["input_value"],
-                expected_value=parameters["expected_output_value"],
-                dtype=tf.uint8,
-            )
-
-    def test_input_values_outside_of_specified_range_are_clipped(self):
-        solarization = Solarization()
+    def _test_input_output(self, layer, input_value, expected_value, dtype):
+        dummy_input = tf.ones(shape=(2, 224, 224, 3), dtype=dtype) * input_value
```

Comment:
Not sure why this is called dummy input, seems to be normal input.

Reply:
yeah-  good point updated.